### DISCUSSION TITLE
Opening SQL files from IFS Browser

### DISCUSSION THREAD

**Discussion by Gregory1701** [codefori/vscode-ibmi#2388] [IBM i] [answered]
Okay, I thought I had this working, but now it's not. I've migrated all of my saved SQL files from my local drive to my IFS subdirectory. 

Test 1: Opening a file I migrated, (single word | dot | 'sql') named SYSTOOLS.sql:

```
/home/PGMGREGORY: DOES_THIS_WORK="YESITDOES" export DOES_THIS_WORK && /usr/bin/attr -p /home/pgmgregory/sql files/SYSTOOLS.sql CREATE_TIME MODIFY_TIME DATA_SIZE OBJTYPE
/home/PGMGREGORY: test -e /home/pgmgregory/sql\ files/SYSTOOLS.sql
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/home/PGMGREGORY: DOES_THIS_WORK="YESITDOES" export DOES_THIS_WORK && /usr/bin/attr -p /home/pgmgregory/sql files/SYSTOOLS.sql CREATE_TIME MODIFY_TIME DATA_SIZE OBJTYPE
{
    "code": 1,
    "signal": null,
    "stdout": "",
    "stderr": "attr: 001-2249 Attribute FILES/SYSTOOLS.SQL is not valid."
}

```

Test 2: Opening a file I migrated, (multiple words | dot | 'sql') named Unused Objects.sql

```
/home/PGMGREGORY: test -w /home/pgmgregory/sql\ files/Unused\ Objects.sql
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/home/PGMGREGORY: test -e /home/pgmgregory/sql\ files/Unused\ Objects.sql
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

```

Test 3: Opening a single word SQL file I created within DB2 for i and saved to my IFS folder named SYSFuncs.sql

```
/home/PGMGREGORY: test -w /home/pgmgregory/sql\ files/SYSFuncs.sql
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/home/PGMGREGORY: test -e /home/pgmgregory/sql\ files/SYSFuncs.sql
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/home/PGMGREGORY: DOES_THIS_WORK="YESITDOES" export DOES_THIS_WORK && /usr/bin/attr -p /home/pgmgregory/sql files/SYSFuncs.sql CREATE_TIME MODIFY_TIME DATA_SIZE OBJTYPE
{
    "code": 1,
    "signal": null,
    "stdout": "",
    "stderr": "attr: 001-2249 Attribute FILES/SYSFUNCS.SQL is not valid."
}

Test 4: Opening a multiple word SQL file I created while working with Adam recently named testing AI with Adam.sql
/home/PGMGREGORY: test -w /home/pgmgregory/sql\ files/testing\ AI\ with\ Adam.sql
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/home/PGMGREGORY: test -e /home/pgmgregory/sql\ files/testing\ AI\ with\ Adam.sql
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/home/PGMGREGORY: DOES_THIS_WORK="YESITDOES" export DOES_THIS_WORK && /usr/bin/attr -p /home/pgmgregory/sql files/testing AI with Adam.sql CREATE_TIME MODIFY_TIME DATA_SIZE OBJTYPE
{
    "code": 1,
    "signal": null,
    "stdout": "",
    "stderr": "attr: 001-2249 Attribute FILES/TESTING is not valid.\nattr: 001-2249 Attribute AI is not valid.\nattr: 001-2249 Attribute WITH is not valid.\nattr: 001-2249 Attribute ADAM.SQL is not valid."
}
```

**Comment by worksofliam** [comment]
@Gregory1701 

1. Connect and reload for me and if it asks you about bash, set it as your default. Initial glance is that it is not your default shell.
2. Looks like there is an improvement for us to make here when it comes to files with spaces.

Moral of the story, spaces in the name likely has additional issues. We just reworked our encoding logic this week so I suspect it had an impact here.

**Reply by Gregory1701** [reply]
@worksofliam When I reconnect and reload I see two interesting things.

1. 'Select default Shell' doesn't show up in my terminal dropdown. My default looks like is currently powershell. How do I set it to bash?
2. The SQL file it previously couldn't open, it opened, but has the name showing as a strikethrough.
![image](https://github.com/user-attachments/assets/ac43fa01-0819-447a-bfba-0ea76349e9b6)

**Reply by worksofliam** [reply]
Per number one, I meant disconnect from your server, then right click on it and connect and reload.

**Comment by worksofliam** [comment]
@Gregory1701 I did find and fix a bug here. PR is here: https://github.com/codefori/vscode-ibmi/pull/2390

**Reply by worksofliam** [reply, marked as answer]
@Gregory1701  just released 2.14.4 which may solve some of the issues here.

**Reply by Gregory1701** [reply]
Thanks @worksofliam ! I will get it installed and test it out. I think it was also complaining about my subdirectory 'SQL files' which has a blank in the name. Did you fixes address this as well?

Side question: In our environment, I have to drop my VPN connection to get the extension updates. Super annoying. Any guidance on what I need to tell my network guys to allow this while on the VPN?

**Reply by worksofliam** [reply]
@Gregory1701 Yes, I believe it will fix all paths with spaces in them.

Regarding the network issue, no idea sadly. Something related to https://marketplace.visualstudio.com/vscode I suspect.

**Reply by Gregory1701** [reply]
This is working much much better, thanks @worksofliam!

**Comment by Gregory1701** [comment]
2.14.4 corrected the issues.
